### PR TITLE
Hide authentication data in PAYG UI (bsc#1199679)

### PR DESCRIPTION
**Pull request by rjmateus**
Signed-off-by: Ricardo Mateus <rmateus@suse.com>

## What does this PR change?

For security reasons, we should hide the authentication data on the webUI. 
Also in the create form, password fields should be of type password.

When editing the ssh connection user should re-provide the authentication data, otherwise, the existing one will be removed.

## GUI diff

Before:

image

image

image

After:

image

image

image

- [ ] **DONE**

## Documentation
- No documentation needed: We are hiding data on the details page, and how to change data is not documented.

- [ ] **DONE**

## Test coverage
- tests already exists and were adapted to this change

- [ ] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/17903
Tracks:
- needs 4.2 backport

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by rjmateus** [comment]
ping @admd to check the prints

File: web/html/src/manager/admin/payg-shared/sshData/payg-ssh-data-view.tsx [hunk 1]
```diff
@@ -24,17 +25,17 @@ const PaygShhDataView = (props: Props) => {
       </dl>
       <dl className="row">
         <dt className="col-xs-2">{t("Password")}</dt>
-        <dd className="col-xs-10">{props.isInstance ? props.payg.password : props.payg.bastion_password}</dd>
+        <dd className="col-xs-10">{props.isInstance ? (props.payg.password || passHidden) : (props.payg.bastion_password || passHidden )}</dd>
```
Comment (Etheryte) [inline web/html/src/manager/admin/payg-shared/sshData/payg-ssh-data-view.tsx:28]: Alternatively `(props.isInstance ? props.payg.password : props.payg.bastion_password) || passHidden`, but I'm fine with either way.

File: web/html/src/manager/admin/payg/payg.tsx [hunk 2]
```diff
@@ -27,6 +27,8 @@ export type paygProperties = {
   bastion_password: string;
   bastion_key: string;
   bastion_key_password: string;
+  instance_edit: bool
```
Comment (Etheryte) [inline web/html/src/manager/admin/payg/payg.tsx:30]: These should be `boolean`.

**Review by Etheryte — APPROVED** [review]
Frontend looks good, thanks for fixing this. 👍

If I'm not mistaken, there should be some small autofixable linting errors, the Github actions don't run for that yet since there's conflicts, but please be sure to fix those before merging. Running `yarn lint` should be enough to fix them.

**Review by srbarrios — APPROVED** [review]
Test changes LGTM

File: web/html/src/manager/admin/payg/payg.tsx (hunk 2 above)
Comment (rjmateus) [reply web/html/src/manager/admin/payg/payg.tsx]: solved

**Comment by rjmateus** [comment]
@juliogonzalez this is a backport and was already released in suma 4.2.

**Pull request merged**